Applications/Utils/ModelPreparation/PartitionMesh/NodeWiseMeshPartitioner.cpp

```diff
@@ -539,6 +539,68 @@ void NodeWiseMeshPartitioner::renumberBulkNodeIdsProperty(
     }
 }
 
+void NodeWiseMeshPartitioner::renumberBulkElementIdsProperty(
+    MeshLib::PropertyVector<std::size_t>* const bulk_element_ids_pv,
+    std::vector<Partition> const& local_partitions) const
+{
+    if (bulk_element_ids_pv == nullptr)
+    {
+        return;
+    }
+
+    auto& bulk_element_ids = *bulk_element_ids_pv;
+
+    std::size_t offset = 0;  // offset in property vector for current partition
+
+    assert(_partitions.size() == local_partitions.size());
+    int const n_partitions = static_cast<int>(_partitions.size());
+    for (int partition_id = 0; partition_id < n_partitions; ++partition_id)
+    {
+        auto const& bulk_partition = _partitions[partition_id];
+        auto const& local_partition = local_partitions[partition_id];
+
+        // Create global-to-local element id mapping for the bulk partition.
+        std::map<std::size_t, std::size_t> global_to_local;
+        {
+            auto const& regular_bulk_elements = bulk_partition.regular_elements;
+            auto const n_regular_bulk_elements = regular_bulk_elements.size();
+            for (std::size_t e = 0; e < n_regular_bulk_elements; ++e)
+            {
+                global_to_local[regular_bulk_elements[e]->getID()] = e;
+            }
+            auto const& ghost_bulk_elements = bulk_partition.ghost_elements;
+            auto const n_ghost_bulk_elements = ghost_bulk_elements.size();
+            std::map<std::size_t, std::size_t> global_to_local;
+            for (std::size_t e = 0; e < n_ghost_bulk_elements; ++e)
+            {
+                global_to_local[ghost_bulk_elements[e]->getID()] =
+                    n_regular_bulk_elements + e;
+            }
+        }
+
+        // Renumber the local bulk_element_ids map.
+        {
+            auto const& local_regular_elements =
+                local_partition.regular_elements;
+            auto const n_local_regular_elements = local_regular_elements.size();
+            for (std::size_t e = 0; e < n_local_regular_elements; ++e)
+            {
+                bulk_element_ids[offset + e] =
+                    global_to_local[bulk_element_ids[offset + e]];
+            }
+            auto const& local_ghost_elements = local_partition.ghost_elements;
+            auto const n_local_ghost_elements = local_ghost_elements.size();
+            for (std::size_t e = 0; e < n_local_ghost_elements; ++e)
+            {
+                bulk_element_ids[offset + n_local_regular_elements +
+                                 e] = global_to_local
+                    [bulk_element_ids[offset + n_local_regular_elements + e]];
+            }
```

Comment:
The loops for the regular and the ghost elements look very similar. Maybe a lambda can be called twice?